src/extlibs/er_force_sim/src/amun/simulator/simrobot.cpp

```diff
@@ -650,6 +650,40 @@ void SimRobot::update(SSLProto::SSL_DetectionRobot *robot, float stddev_p,
     m_lastSendTime = time;
 }
 
+bool SimRobot::touchesBall(SimBall *ball) const
+{
+    if (m_holdBallConstraint)
+    {
+        return true;
+    }
+
+    int numManifolds = m_world->getDispatcher()->getNumManifolds();
+    for (int i = 0; i < numManifolds; ++i)
+    {
```

Comment:
please explain this better with some comments

Reply:
added